pkg/commands/git_commands/tag.go

```diff
@@ -27,13 +29,38 @@ func (self *TagCommands) CreateAnnotated(tagName, ref, msg string) error {
 	return self.cmd.New(cmdArgs).Run()
 }
 
-func (self *TagCommands) Delete(tagName string) error {
+func (self *TagCommands) LocalDelete(tagName string) error {
 	cmdArgs := NewGitCmd("tag").Arg("-d", tagName).
 		ToArgv()
 
 	return self.cmd.New(cmdArgs).Run()
 }
 
+func (self *TagCommands) RemoteDelete(tagName string) error {
+	cmdArgs := NewGitCmd("push").Arg("--delete", "origin", tagName).
```

Comment:
Hang on, we should be able to get the upstream remote and remote branch from the branch itself, so there's no need for the user to specify the remote!

Reply:
@jesseduffield  I've managed to retrieve the current branch using the `GetCheckedOutRef` method from the `RefsHelper` struct. Is this what you were thinking about?

I'm not 100% sure, but if you can derive the tag upstream remote from the branch while deleting it, shouldn't you be able to do the same while pushing it? In contrast, if you can choose a different upstream for a tag, couldn't it be different from the branch one?

This feature is obviously much more straightforward for the branch controller.

Reply:
GetCheckedOutRef won't work because we want the currently selected branch, so depending on where you are it would be `self.c.Contexts().Branches.GetSelected()`

I'm not sure what you mean re: pushing a branch. If we already have an upstream for the branch it shouldn't prompt you for the upstream when you go to push.

I'm also not sure what you mean regarding the tag. Tags don't maintain an upstream like branches do.

Reply:
Sorry @jesseduffield, I misspoke - let me better clarify my doubt.

Your comment suggested inferring the remote for tags (on deletion) using the information of the branch it belongs to. The relationship between the remote on which I perform the tag storage and the upstream remote of the branch it references is unclear to me.

The user can push the branch and the tag to different remotes, right?

Reply:
Sorry I meant that we can infer the remote from the branch. As for tags, tags aren't associated to branches or upstream so we would need to show the remote prompt for tags

Reply:
Alright, thanks! 🙏🏻